Shift3/boilerplate-client-react
# feat(dependencies): Added project dependencies

## Changes
Added the following production dependencies:

1. redux, react-redux, redux-devtools, redux-thunk
2. styled-components
3. react-bootstrap
4. axios
5. source-map-explorer
6. sentry
7. dotenv

## Purpose
To optimize project with corresponding libraries.

## Approach
The project did not have any of these libraries before, and now will have the following libraries available for the rest of the build process:
1. Redux was brought in as a comprehensive library to handle state management. For more info 'https://redux.js.org/'
2. Styled-components and React-Bootstrap for handling all styling of components. For more info 'https://react-bootstrap.github.io/' and 'https://styled-components.com/'
3. React-router-dom for page routing 'https://reactrouter.com/web/guides/quick-start'
4. Axios for a more efficient way to work with the api 'https://www.npmjs.com/package/axios'
5. Source-map-explorer to analyze code size 'https://www.npmjs.com/package/source-map-explorer'
6. Sentry for error tracking 'https://sentry.io/for/javascript/'
7. Dotenv for loading environment variables 'https://www.npmjs.com/package/dotenv' 

## Learning
Some time was spent on determining the best and most efficient libraries for production build, as well as determining which dev dependencies were also required. All related development dependencies will be added in an alternate PR.

## Testing
1. Pull in the changes to your local copy of this branch.
2. Run [`yarn start`] and make sure the app runs with no issue.

### Closes #39
